lib/generator.ts

```diff
@@ -542,22 +570,83 @@ export class Generator extends Cli {
     return source
   }
 
+  private sourceLines(source: string) {
+    source = source.replace(/\r\n/g, '\n')
+    source = source.replace(/\n\r/g, '\n')
+    source = source.replace(/\r/g, '\n')
+
+    const lines = source.split('\n')
+    return lines
+  }
+
+  private findDependencies(
+    typings: IDeclarationMap,
+    moduleName: string,
+    shakenTypings: Set<string>,
```

Comment:
Please avoid modifying argument by reference - it causes code which is very hard to read.

It would be best to construct a Set inside findDependencies. Since function is recursive, it would require merging sets but it can be easily done.

```
let a = new Set([1, 2, 3])
let b = new Set([1, 2, 3, 4])
let c = new Set([...a, ...b])
// c now contains: Set(4) {1, 2, 3, 4}
```